0000-catalog.md

```diff
@@ -0,0 +1,779 @@
+consumed by the grid namespace prefix, the catalog prefix, and gs1_company_prefix
+there
+are 52 hex characters remaining in the address. Then the 15 digits of the
+catalog_id can
+be added with a left padding of 35-hex-character zeroes, and that string is right 
+padded with 2-hex-character zeroes to accommodate potential future storage 
+associated with the GS1 Catalog representation, for example:
+
+```
+"621dee" + "03" + "aaaa123456" + "00000000000000000000000000000000000" + catalog_id 
++ "00" // catalog_id == hash(catalog_name).trunc(15)
+```
+
+Using SHA-2 from the rust-crypto crate we can tie the catalog_id to catalog_name
+to prevent duplication of catalogs when invoking the catalog_create action.
+
+```
+extern crate crypto;
+use self::crypto::digest::Digest;
+use crypto::sha2::Sha512;
+
+fn sha2_512() {
+    // create a SHA2-512 hasher
+    let mut hasher = Sha512::new();
+
+    // write input message
+    hasher.input_str("catalog_name");
+
+    // read hash digest
+    let hex = hasher.result_str();
+    let hex_15 = &hex[0..15];
+
+    println!("SHA2-512 Hash is {:?}", hex);
+    println!("SHA2-512 Hash (15) {:?}", hex_15);
+}
+
+>> SHA2-512 Hash is:
+cad4e781e7977123893a42c7e7230e798bf5693f95c04980ba70bbe3504b194227c62e7e32d83ddb
+d083979720d7025b2cc9f68ac7b572dc6a607409b2dcdc43
+>> SHA2-512 Hash (15): cad4e781e797712
+```
+
+The full catalog address would look like:
+
+```
+"621dee03aaaa12345600000000000000000000000000000000000cad4e781e79771200"
+```
+
+### Referencing a Catalog Product
+
+Catalogs products are uniquely referenced by a composite key composed of their
+catalog_id and product_id. For example:
+
+```
+get_catalog_product(catalog_id, product_id) // hash(catalog_name), product_id
+set_catalog_product(catalog_id, product_id, catalog_product) //hash(catalog_name), product_id, catalog_product
+```
+
+### Catalog Product Addressing in the Merkle-Radix State System
+
+The Grid GS1 Catalog Product state address will be prefixed with the Grid
+namespace of "621dee", in addition to "02" (product namespace), "01" for (gs1
+product namespace), and the catalog_id.
+
+Therefore, all addresses starting with:
+
+```
+"621dee" + "02" + "01 + "cad4e781e797712" // catalog_id
+```
+
+are Grid GS1 Catalog Products and are expected to contain the standard
+attributes of a Grid GS1 Product as well as the extra PropertyValues that
+are defined in the catalog product schema. These PropertyValues are in
+addition to the expected PropertyValues of a [Grid GS1
+Product](https://github.com/hyperledger/grid-rfcs/blob/fbedec06d70b16492fea9f6b1e87146c5fc56771/0000-product.md).
+
+GTIN formats consist of 14-digit "numeric strings" which include some amount of
+internal "0" padding depending on the specific GTIN format (GTIN-12, GTIN-13, or 
+GTIN-14). After the 12-hex-characters that are consumed by the gridnamespace 
+prefix, the product namespace prefix, GS1 namespace prefix, and catalog product 
```

Comment:
catalog product namespace should be replaced with catalog id and 12 should be to include the number of hex in "621dee" + "02" + "01 + "cad4e781e797712". (or just remove catalog product namespace  and say it is 10 hex and explain adding the catalog id after)

Reply:
Fixed, also adjusted the number ranges for the remaining characters in the address. There were slightly off. ✅